modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMTemplateLocalServiceImpl.java

```diff
@@ -396,11 +398,16 @@ public List<DDMTemplate> copyTemplates(
 		List<DDMTemplate> oldTemplates = ddmTemplatePersistence.findByC_C_T(
 			classNameId, oldClassPK, type);
 
+		Indexer<DDMTemplate> indexer = IndexerRegistryUtil.nullSafeGetIndexer(
+			DDMTemplate.class);
+
 		for (DDMTemplate oldTemplate : oldTemplates) {
 			DDMTemplate newTemplate = copyTemplate(
 				userId, oldTemplate, newClassPK, oldTemplate.getNameMap(),
 				oldTemplate.getDescriptionMap(), serviceContext);
 
+			indexer.reindex(newTemplate);
+
 			newTemplates.add(newTemplate);
 		}
 
```

Comment:
Hi @sontruongces,

I think we could change and pass the list of newTemplates instead of re-indexing one by one:

```
Indexer<DDMTemplate> indexer = IndexerRegistryUtil.nullSafeGetIndexer(
	DDMTemplate.class);
		
indexer.reindex(newTemplates);
```

Reply:
@carolmariaabb, thanks for your response. I updated the fix and tested success in my local. Thanks!